### DISCUSSION TITLE
.NET 7.0?

### DISCUSSION THREAD

**Discussion by michaelcsikos** [MarimerLLC/csla#3218] [General]
".NET 7 will be supported for 18 months as a standard-term support (STS) release (previously known as a current release)." Are than any compelling reasons to migrate, and when will CSLA support it?

**Comment by rockfordlhotka** [comment]
CSLA 6 works with .NET 7 today.

CSLA 7 will _target_ .NET 7, but that's a minor formality, since no code changes are required for that to happen. However, CSLA 7 will include some breaking changes because it is a major version update, and I'm not going to rush those breaking changes out the door - especially since CSLA 6 works with .NET 7 today.

In terms of compelling reasons to migrate to .NET 7, I think there are a couple

1. Don't fall behind - it'll be easier to go from 7 to 8 than 6 to 8 based in historical experience
2. Microsoft put a lot of work into performance in .NET 7, and they've published some blog posts about it; specially for aspnetcore there are some major improvements apparently
3. It sounds like they did a lot of stability work in MAUI, so if you are using MAUI it is probably a worthy upgrade
4. They continued to improve Blazor with some features, and some improvements around AOT (ahead of time compilation), which could be compelling

Mostly it seems like points 1 and 2 are the primary drivers imo.

**Reply by michaelcsikos** [reply]
1,2 and 4 are good reasons for us. Thanks for the quick response.

**Comment by kcabral817** [comment]
Falling behind is muy bad.  I just spent the last month upgrading .net framework to core and ef to ef core.  What a nightmare.

**Reply by rockfordlhotka** [reply]
That was a big jump though - because modern .NET is basically a reimplementation of the whole platform, just based on .NET Framework.

Moving from 3.1 to 6 takes some real effort, but 5 to 6 and 6 to 7 have been relatively easy. Hopefully that trend continues!

**Comment by Chicagoan2016** [comment]
Does it mean we don't have to target library projects to .NET Standard, Classic UI to .NET Framework 4.XX etc.?
:)

**Reply by rockfordlhotka** [reply]
Absolutely! If you are creating a class library that is only going to be used by a .NET 5 or higher app, then you have been able to target that version of .NET in your class library.

The restriction around netstandard2.0 isn't a CSLA thing, it is that if you want your class library to be used by _both_ a .NET Framework (or WinRT or Xamarin) app and also a "modern" app, then you must target ns2.0.

But if you are only creating a modern .NET app, and your class library isn't used by any legacy apps, then just target the current version of .NET.

**Reply by Chicagoan2016** [reply]
@rockfordlhotka  Thank you Sir, we have some Winforms applications and would love to target all projects (in our VS solution) to .NET 7 in the next iteration.
(I know we should move away from Winforms but that's a story for another day :))

**Reply by Chicagoan2016** [reply]
I created a Winforms application in VS 2022, targeting .NET 7 and it works!!!!!!

**Reply by michaelcsikos** [reply]
WinForms isn't dead yet!

**Reply by Chicagoan2016** [reply]
@michaelcsikos  yes and MVVM in winforms is right around the corner
https://devblogs.microsoft.com/dotnet/winforms-enhancements-in-dotnet-7/

**Reply by rockfordlhotka** [reply]
If you aren't beholden to legacy code, you can use DI (and thus MVVM) in WinForms and WPF today. The real blocker is the massive amounts of legacy code in existing apps that can't be easily converted to a DI/MVVM model.

Yes, MVVM without DI requires extra support/work, but _with_ DI, MVVM is extremely smooth., and you can use DI with WinForms as long as legacy code isn't an issue.

Just change `Program.cs` to create the first form via DI, and thereby any subsequent forms are created via DI - which means that a form can require a viewmodel via its constructor, and all of a sudden you are writing WinForms code that is comparable to aspnetcore or Blazor. It is really smooth!

**Reply by rockfordlhotka** [reply]
I just double-checked, and the Windows Forms example in the main branch is a basic example of this:

https://github.com/MarimerLLC/csla/tree/main/Samples/WinFormsExample/WinFormsExample
